packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -146,25 +177,67 @@ export const DatePicker: React.VFC<DatePickerProps> = ({
     disabledAfter
   )
 
-  const handleOnDayChange = (day: Date, modifiers: DayModifiers): void => {
-    if (isDisabledDate(day, disabledDays)) {
-      return
+  const handleDayChange = (
+    date: Date | undefined,
+    inputValue?: string
+  ): void => {
+    const baseResponse = {
+      date,
+      inputValue,
+      status: undefined,
+      validationMessage: undefined,
+      isInvalid: false,
+      isDisabled: false,
+      isEmpty: false,
+      isValidDate: false,
     }
-    onDayChange(day)
-    setIsOpen(false)
+
+    if (date === undefined) {
+      onValidate({
+        ...baseResponse,
+        isEmpty: true,
+      })
+      return onDayChange(undefined)
```

Comment:
I wonder if instead of having two callbacks, if you considered having a second argument to `onDayChange` containing the validation information?

Reply:
Yes we considered it, however when we were building it out we found instances where we would want to run the validation and not update the day (currently in the initial render, but there were previous changes too).

In saying that, we discussed a next iteration where we might handle the inbuilt validation without the consumer needing to create state for validation should they not have any custom validation themselves. If we find that both functions will always be called together, then we could consider merging them into one (since there isn't much usage of the component yet).